providers/imap/src/main/java/com/sun/mail/imap/package-info.java

```diff
@@ -333,6 +333,22 @@
  * </TR>
  *
  * <TR>
+ * <TD><A ID="mail.imap.executor.writetimeout">mail.imap.executor.writetimeout</A></TD>
+ * <TD>java.util.concurrent.ScheduledExecutorService</TD>
+ * <TD> Provides specific ScheduledExecutorService for mail.imap.writetimeout option.
+ * The value of mail.imap.writetimeout shouldn't be a null.
+ *  For provided executor pool it is highly recommended to have set up in true
+ * {@link java.util.concurrent.ScheduledThreadPoolExecutor#setRemoveOnCancelPolicy(boolean)}.
+ * Without it write methods will create garbage that would only be reclaimed after the timeout.
+ * Do not call {@link java.util.concurrent.ScheduledThreadPoolExecutor#shutdownNow()} in your executor,
+ * it can kill the running task.
```

Comment:
It would be ok to use shutdownNow when JavaMail sockets are closed.  This would be all service subclasses (Store/Transport).  Invoking run on the returned Future objects would force close the open connections.

We'll have to think of how to capture that idea a bit better.

Reply:
Fixed with the provided suggestion